src/data/make_dataset.py

```diff
@@ -18,6 +24,188 @@ def main(input_filepath, output_filepath):
     logger.info("making final data set from raw data")
 
 
+def fetch_all_tests(raw_data, status_code, test_name):
+    """fetch_all_tests function takes raw data and status code of the test.
+
+    Returns a dataframe with all the tests with that status code.
+    """
+    # Fetch the list of all tests
+    tests_list = testgrid_labelwise_encoding(
+        raw_data, status_code, test_name, overall_only=False
+    )
+    # Convert to dataframe
+    tests_df = pd.DataFrame(
+        tests_list,
+        columns=[
+            "timestamp",
+            "tab",
+            "grid",
+            "test",
+            "test_duration",
+            "failure/passing",
+        ],
+    )
+    # We will drop all the rows having NaN values
+    tests_df = tests_df.dropna()
+    tests_df = tests_df[tests_df["failure/passing"]]
+    return tests_df
+
+
+def decode_run_length(x):
+    """Decode the run length encoded data into an unrolled form.
+
+    Returns a list of values.
+
+    E.g. takes in [{"value":12, "count":3}, {"value":1, "count":2}]
+    and gives [12, 12, 12, 1, 1]
+    """
+    lst = []
+    for run_length in x:
+        extension = [run_length["value"]] * run_length["count"]
+        lst.extend(extension)
+    return lst
+
+
+def testgrid_labelwise_encoding(data, label, test_name, overall_only=True):
```

Comment:
can we use this [file](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/notebooks/data-sources/TestGrid/metrics/metric_template.ipynb) or is it not possible to import that here?

Reply:
I don't think we could do that here.